control/gps.py

```diff
@@ -31,7 +38,8 @@ def read(self):
         """Returns a GpsReading object with the values supplied"""
         msg = None
         # TODO: Add try except chain to catch any dirty GPS output exceptions
+        # TODO: Add timeout for reads using signal library
```

Comment:
Still think we might be able to just set the timeout of pyserial... but we might wanna catch it here and package it into a specific GPS exception so its obvious that we timed out from the GPS serial read. Not familiar with signal that might be a better approach but we'll have to investigate.